packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -1945,6 +2008,36 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
     }
   }
 
+  /// Determines whether or not to enable spell check and if enabled, the
+  /// [SpellCheckConfiguration] that will be used.
+  ///
+  /// If spell check is enabled, this will try to infer a value for
+  /// [SpellCheckService] if left unspecified and initialize the
+  /// [SpellCheckSuggestionsHandler] used to mark misspelled words and display
+  /// suggestions.
+  SpellCheckConfiguration? _inferSpellCheckConfiguration(SpellCheckConfiguration? configuration) {
+    if (configuration == null || configuration == const SpellCheckConfiguration.disabled()) {
+      // Spell check is disabled.
+      return null;
+    }
+
+    _spellCheckEnabled = true;
+    SpellCheckService? spellCheckService = configuration.spellCheckService;
+
+    assert(
+      spellCheckService != null
+      || WidgetsBinding.instance.platformDispatcher.nativeSpellCheckServiceDefined,
+      'spellCheckService must be specified for this platform because no default service available',
+    );
+
+    spellCheckService = spellCheckService ?? DefaultSpellCheckService();
```

Comment:
Similar to a comment I left above about _spellCheckEnabled, I wonder if there's a way to organize this so that there is only one source of truth.  As it is now, the instance variable `spellCheckService` could possibly be out of sync with widget.spellCheckConfiguration if _inferSpellCheckConfiguration hasn't been called recently.

Maybe a spellCheckConfiguration getter (that utilizes caching if necessary)?

Reply:
I totally agree with changing _spellCheckEnabled to just use a getter, but this is not an instance variable. It's just a variable used in this method. So, changing _spellCheckEnabled should reduce it down to one source of truth!

Reply:
Oh got it, I probably misread it.  I'm glad that making it a getter should work out though.